src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -225,119 +248,183 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
+                ths.forEach(c => this.setCellWidth(c, 0));
+            }
+        });
+    }
+
+    resizeFragmentTable = (table: HTMLElement | null, width: string) => {
+        if (!table) {
+            return;
         }
 
-        this.updateStylesheet();
+        table.style.width = width;
+    }
 
-        getScrollbarWidth().then((scrollbarWidth: number) => setState({ scrollbarWidth }));
+    forceHandleResize = () => this.handleResize();
 
+    handleResizeIf = memoizeOne((..._: any[]) => {
         const { r0c0, r0c1, r1c0, r1c1 } = this.refs as { [key: string]: HTMLElement };
 
+        r0c1.style.marginLeft = '';
+        r1c1.style.marginLeft = '';
+        r0c0.style.width = '';
+        r1c0.style.width = '';
 
-        // Adjust [fixed columns/fixed rows combo] to fixed rows height
-        let trs = r0c1.querySelectorAll('tr');
-        Array.from(r0c0.querySelectorAll('tr')).forEach((tr, index) => {
-            const tr2 = trs[index];
+        const r0c0Table = r0c0.querySelector('table');
+        const r0c1Table = r0c1.querySelector('table');
+        const r1c0Table = r1c0.querySelector('table');
 
-            tr.style.height = `${tr2.clientHeight}px`;
-        });
+        if (r0c0Table) { r0c0Table.style.width = ''; }
+        if (r0c1Table) { r0c1Table.style.width = ''; }
+        if (r1c0Table) { r1c0Table.style.width = ''; }
 
-        // Adjust fixed columns headers to header's height
-        let trths = r1c1.querySelectorAll('tr > th:first-of-type');
-        Array.from(r1c0.querySelectorAll('tr > th:first-of-type')).forEach((th, index) => {
-            const tr2 = trths[index].parentElement as HTMLElement;
-            const tr = th.parentElement as HTMLElement;
+        this.resetFragmentCells(r0c0);
+        this.resetFragmentCells(r0c1);
+        this.resetFragmentCells(r1c0);
 
-            tr.style.height = getComputedStyle(tr2).height;
-        });
+        this.handleResize();
+    });
 
-        if (fixed_columns) {
-            const r1c0Table = r1c0.querySelector('table') as HTMLElement;
-            const r1c1Table = r1c0.querySelector('table') as HTMLElement;
+    handleResize = (previousWidth: number = NaN, cycle: boolean = false) => {
+        const {
+            fixed_columns,
+            fixed_rows,
+            setState
+        } = this.props;
 
-            r1c0Table.style.width = getComputedStyle(r1c1Table).width;
+        const { r1c1 } = this.refs as { [key: string]: HTMLElement };
+        const r1c1Table = r1c1.querySelector('table') as HTMLElement;
 
-            const lastVisibleTd = r1c0.querySelector(`tr:first-of-type > *:nth-of-type(${fixed_columns})`);
+        const currentTableWidth = getComputedStyle(r1c1Table).width;
 
-            let it = 0;
-            let currentWidth = r1c0.getBoundingClientRect().width;
-            let lastWidth = currentWidth;
+        // Not pixels -> not displayed
+        if (!/\d+px/.test(currentTableWidth)) {
+            return;
+        }
```

Comment:
If not `px`, the table is not rendered, don't resize it. Lucky https://github.com/plotly/plotly.js/pull/4925 happened at the same time, otherwise it would have gone unoticed.

Reply:
OIC, `'100.1px'` will in fact pass this test as it's not a whole string match. Worried me at first but it should work.